github/resource_github_actions_secret.go

```diff
@@ -0,0 +1,150 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/google/go-github/v29/github"
+	"github.com/hashicorp/terraform/helper/schema"
+	"golang.org/x/crypto/nacl/box"
+	"log"
+)
+
+func resourceGithubActionsSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubActionsSecretCreateOrUpdate,
+		Read:   resourceGithubActionsSecretRead,
+		Update: resourceGithubActionsSecretCreateOrUpdate,
+		Delete: resourceGithubActionsSecretDelete,
+
+		Schema: map[string]*schema.Schema{
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"secret_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"plaintext_value": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"key_id": {
```

Comment:
Perhaps it's better to not have these arguments and instead just fetch and use the public key on create?

Reply:
It's a good shout. I initially provided these to allow users to determine the key / key id without the need for `github_actions_secrets_public_key` but I think just not including them as you suggest is better.